src/cosim/uri.cpp

```diff
@@ -559,4 +559,20 @@ cosim::filesystem::path file_uri_to_path(const uri& fileUri)
 }
 
 
+cosim::filesystem::path file_query_uri_to_path(
+    const cosim::uri& baseUri,
+    const cosim::uri& queryUri)
+{
+    const auto query = queryUri.query();
+    if (query && query->find("file=") < query->size()) {
+        if (query->find("file=file:///") < query->size()) {
+            return file_uri_to_path(std::string(query->substr(5)));
+        }
+        const auto uriToAppend = cosim::path_to_file_uri(cosim::file_uri_to_path(baseUri).parent_path());
```

Comment:
I agree with the reasoning and if this was an isolated case it would make sense to rename it to `baseUriParentPath` or similar. The reason I am still leaning towards keeping the current code is that `baseUri` is consistently used in multiple similar functions and I find it strange to have a one-off parameter that is very similar but not quite equal. Further inputs are much appreciated

Reply:
If this would be an improvement to readability and/or usability of the code, then I'd say just do it. No need to rely on adherence to a similar implementation elsewhere (in fact, perhaps that too should be updated).